app/1.0.x/upgrading.md

```diff
@@ -0,0 +1,307 @@
+upstream services (via `/apis`) instead of using Routes for endpoints (via
+`/routes`) and Services for upstream services (via `/services`), now is the
+time to do so. Kong 1.0 will refuse to run migrations if you have any entity
+configured using `/apis` in your datastore. Create equivalent Routes and
+Services and delete your APIs. (Note that Kong does not do this automatically
+because the naive option of creating a Route and Service pair for each API
+would miss the point of the improvements brought by Routes and Services;
+the ideal mapping of Routes and Services depends on your microservice
+architecture.)
+
+If you use additional plugins other than the ones bundled with Kong,
+make sure they are compatible with Kong 1.0 prior to upgrading.
+See the section above on Plugins for information on plugin compatibility.
+
+### Migration Steps from 0.14
+
+Kong 1.0 introduces a new, improved migrations framework.
+It supports a no-downtime, Blue/Green migration model for upgrading
+from 0.14.x. The full migration is now split into two steps,
+which are performed via commands `kong migrations up` and
+`kong migrations finish`.
+
+For a no-downtime migration from a 0.14 cluster to a 1.0 cluster,
+we recommend the following sequence of steps:
+
+1. Download 1.0, and configure it to point to the same datastore
+   as your 0.14 cluster. Run `kong migrations up`.
+2. Both 0.14 and 1.0 nodes can now run simultaneously on the same
+   datastore. Start provisioning 1.0 nodes, but do not use their
+   Admin API yet. Prefer making Admin API requests to your 0.14 nodes
+   instead.
+3. Gradually divert traffic away from your 0.14 nodes, and into
+   your 1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+4. When your traffic is fully migrated to the 1.0 cluster,
+   decommission your 0.14 nodes.
+5. From your 1.0 cluster, run: `kong migrations finish`.
+   From this point on, it will not be possible to start 0.14
+   nodes pointing to the same datastore anymore. Only run
+   this command when you are confident that your migration
+   was successful. From now on, you can safely make Admin API
+   requests to your 1.0 nodes.
+
+At any step of the way, you may run `kong migrations list` to get
+a report of the state of migrations. It will list whether there
+are missing migrations, if there are pending migrations (which have
+already started in the `kong migrations up` step and later need to
+finish in the `kong migrations finish` step) or if there
+are new migrations available. The status code of the process will
+also change accordingly:
+
+* `0` - migrations are up-to-date
+* `1` - failed inspecting the state of migrations (e.g. database is down)
+* `3` - database needs bootstrapping:
+  you should run `kong migrations bootstrap` to install on
+  a fresh datastore.
+* `4` - there are pending migrations: once your old cluster is
+  decomissioned you should run `kong migrations finish` (step 5 above).
```

Comment:
Can clarify what is the state of the cluster when the exit code is 4?
As per my understanding, this exit code means that the newer version can now be safely started, the old version can safely run as well but there is another step (finish) that needs to be execute to finish the upgrade process.

We refer to the migrations in this state as pending I find the term a little ambiguous, hence this comment (I don't have a better term myself to suggest).

Reply:
> As per my understanding, this exit code means that the newer version can now be safely started, the old version can safely run as well but there is another step (finish) that needs to be execute to finish the upgrade process.

Yes, your understanding is correct.  Amended based on your description.